blob/blob.go

```diff
@@ -99,14 +106,17 @@ import (
 // It implements io.ReadCloser, and must be closed after
 // reads are finished.
 type Reader struct {
-	b   driver.Bucket
-	r   driver.Reader
-	end func(error) // called at Close to finish trace and metric collection
+	b        driver.Bucket
+	r        driver.Reader
+	end      func(error) // called at Close to finish trace and metric collection
+	provider string      // for metric collection
 }
 
 // Read implements io.Reader (https://golang.org/pkg/io/#Reader).
 func (r *Reader) Read(p []byte) (int, error) {
 	n, err := r.r.Read(p)
+	stats.RecordWithTags(context.Background(), []tag.Mutator{tag.Upsert(oc.ProviderKey, r.provider)},
```

Comment:
WDYT about keeping a counter here and exporting the metric in `Close` instead?

Reply:
That's less accurate (Close might not be called; or the reading might go on for minutes or hours) and less maintainable. Are you concerned about performance? Nothing is getting "exported" (sent off-box) at this point.  This line shouldn't take much time. Unless someone is foolishly reading in very small chunks, it's unlikely to be noticeable. But we could benchmark (against a real service with real buffer sizes).

Reply:
OK, your arguments are convincing :-).